src/components/WithIsFocused.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+import React, { Component as ReactComponent } from "react";
+import type { ComponentType } from "react";
+import type {
+  NavigationEventSubscription,
+  NavigationScreenProp,
+  NavigationState
+} from "react-navigation";
+
+type Props = {
+  navigation: NavigationScreenProp<NavigationState>
+};
+
+type State = {
+  isFocused: boolean
+};
+
+// This component stops children from updating when the given navigation
+// is not inFocus (I.e. not in the state between willFocus and blurred)
+const withIsFocused = <A>(Component: ComponentType<A>) =>
```

Comment:
I still need to add tests for this HOC. Just want to gauge how effective this is on real Android devices before I do.